flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/metrics/WatermarkGauge.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.streaming.runtime.metrics;
+
+import org.apache.flink.metrics.Gauge;
+
+/**
+ * A {@link Gauge} for exposing the current low watermark.
+ */
+public class WatermarkGauge implements Gauge<Long> {
+
+	private long currentLowWatermark = Long.MIN_VALUE;
+
+	public void setCurrentLowWatermark(long watermark) {
+		this.currentLowWatermark = watermark;
```

Comment:
Looking solely at this class alone, I wonder if we should add verification logic here that the set low watermark is ascending.

Reply:
I was hesitant since it would be additional runtime overhead for watermark processing that, as i understand it, should be unnecessary assuming a correct source/watermark assigner implementation.